src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+                                                         null, participantIdentifier, "btn btn-primary btn-xs", 
+                                                         "Moderate Responses");
+            participantPanel = InstructorFeedbackResultsGroupByQuestionPanel.buildInstructorFeedbackResultsGroupByQuestionPanelWithModerationButton(
+                                            participantIdentifier, bundle.getNameForEmail(participantIdentifier),
+                                            questionTables, getProfilePictureLink(participantIdentifier), 
+                                            true, moderationButton);
+        } else {
+            participantPanel = InstructorFeedbackResultsGroupByQuestionPanel.buildInstructorFeedbackResultsGroupByQuestionPanelWithoutModerationButton(
+                                            questionTables, getProfilePictureLink(participantIdentifier), 
+                                            viewType.isFirstGroupedByGiver(), participantIdentifier, bundle.getNameForEmail(participantIdentifier));
+        }
+        
+        return participantPanel;
+    }
+    
+    
+    private void finaliseBuildingSectionPanel(
+                         InstructorFeedbackResultsSectionPanel sectionPanel, String sectionName,
+                         LinkedHashMap<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesGroupedByTeam,
+                         Set<String> teamsWithResponses) {
+        switch (viewType) {
+            case GIVER_QUESTION_RECIPIENT:
+            case RECIPIENT_QUESTION_GIVER:
+                prepareHeadersForTeamPanelsInSectionPanel(sectionPanel);
+                if (responsesGroupedByTeam != null && !responsesGroupedByTeam.isEmpty()) {
+                    buildTeamsStatisticsTableForSectionPanel(sectionPanel, responsesGroupedByTeam, 
+                                                             teamsWithResponses);
+                }
+                sectionPanel.setSectionName(sectionName);
+                sectionPanel.setSectionNameForDisplay(sectionName.equals(Const.DEFAULT_SECTION) ? DISPLAY_NAME_FOR_DEFAULT_SECTION 
+                                                                                                : sectionName);
+                sectionPanel.setDisplayingTeamStatistics(true);
+                break;
+            default:
+                Assumption.fail();
+                break;
+        }
+    }
+    
+
+    private void buildMissingTeamAndParticipantPanelsWithoutModerationButtonForSection(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String section,
+                                    Set<String> receivingTeams) {
+        buildMissingTeamAndParticipantPanelsForSection(sectionPanel, section, receivingTeams, 
+                                                       false);
+    }
+    
+    private void buildMissingTeamAndParticipantPanelsWithModerationButtonForSection(
```

Comment:
Oh, there were multiple boolean parameters initially. It should be possible to remove this entirely and rely on the viewtype for behaviour